_posts/2019-12-08-what-is-a-shell.md

```diff
@@ -0,0 +1,166 @@
+---
+layout: post
+title: "Nushell - The Cross Platform Shell"
+date: 2019-12-10
+categories:
+- rust
+- programming
+- commandline
+---
+### What is a Shell?
+
+A shell is command line (text) or graphical interface that lets you interact with the programs of an operating system.
+The shell program is typically the outermost layer from the operating system kernel, which is where it gets the name "shell".
+If you have ever interacted with MacOS/Linux *terminal* or CMD and PowerShell on Windows, you're using what most users will call "the shell".
+Technically, the program that opens up the window you're typing into is most likley a terminal emulator,
+but the program running inside accepting your input is the shell.
+
+An operating system shell allows you interact with programs installed on the operating system which in turn operate on the file system, 
+system processes, devices that make up the system you're on, and much more.
+In some cases, you are able to communicate with completely different machines as over SSH, aka Secure Shell.
+Software developers and IT professional use shells for many tasks such as batch processing files, searching the file system, running custom scripts, downloading the uploading files to the internet, automating their daily workflows, and even playing games!
+
+Some examples of well known shells are Bash, Zsh, Fish, Powershell, Korn Shell, and csh.
```

Comment:
nit: I know the list isn't meant to be comprehensive, but I feel like it would be good to call out Bourne Shell (`sh`) 🙂

Reply:
whoooops! How embarassing 😆